### PR TITLE
Minor refactoring and warning fix for gcc

### PR DESCRIPTION
**Pull request by biodranik**
Closes #8152

@Ferenc- PTAL

### CODE REVIEW

**Review by Ferenc- — COMMENTED** [review]
I like the `enum class` modernisation.
Only it has a broader impact, so `platform_tests` and `generator_tests` still needs adaptation.
Ultimately what has eliminated the warning was the usage of the always inited `std::ostream` which can be separated from rest of the changes, for a quicker turnaround.

**Comment by biodranik** [comment]
> for a quicker turnaround.

Long-term value from a safer, cleaner code is better than a quicker turnaround. A lot of code was written by us more than a decade ago. Imagine how much we have learned since then, and how better the compiler tools became. A good existing code promotes better code-practices to any new contributors.

**Comment by Ferenc-** [comment]
> Long-term value from a safer, cleaner code is better than a quicker turnaround.

I don't see it as a dicothomy. We can have both in iterations.

File: coding/sha1.cpp [hunk 1]
```diff
@@ -18,14 +18,14 @@ namespace coding
...
-  uint32_t constexpr kFileBufferSize = 8192;
   try
   {
-    base::FileData file(filePath, base::FileData::OP_READ);
+    base::FileData file(filePath, base::FileData::Op::READ);
     uint64_t const fileSize = file.Size();
 
     CSHA1 sha1;
     uint64_t currSize = 0;
+    uint32_t constexpr kFileBufferSize = 8192;
```
Comment (zloidemon) [inline coding/sha1.cpp:28]: А можно добавить описание этой константе? Я вот ещё одну нашел здесь https://github.com/organicmaps/organicmaps/blob/80720f74a30fb3e1ebb521f34dbe88e1511f9802/3party/minizip/minizip.hpp#L13. Может тоже надо переделать?

**Review by vng — APPROVED** [review]

File: coding/sha1.cpp (hunk 1 above)
Comment (biodranik) [reply coding/sha1.cpp:28]: @zloidemon возможно, почему вас заинтересовали именно эти константы?

**Comment by biodranik** [comment]
> I don't see it as a dichotomy. We can have both in iterations.

A compilation warning is not an urgent production issue that requires iterations. Two iterations require 2x more resources to write, test, review, release.

**Comment by Ferenc-** [comment]
> > I don't see it as a dichotomy. We can have both in iterations.
> 
> A compilation warning is not an urgent production issue that requires iterations. Two iterations require 2x more resources to write, test, review, release.

By iterations, I meant PRs.

**Comment by biodranik** [comment]
When the time is extremely limited, the real zen is to do everything only once )

**Pull request merged**